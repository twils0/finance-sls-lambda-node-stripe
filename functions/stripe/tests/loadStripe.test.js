const stripe = require('stripe');
const { getSecret } = require('../../aws/getSecret');

const { loadStripe } = require('../loadStripe');

global.console.log = jest.fn();
jest.mock('../../aws/getSecret', () => ({
  getSecret: jest.fn(),
}));
jest.mock('stripe', () => jest.fn());

const stripeObject = { test: 'testStripe' };
stripe.mockReturnValue(stripeObject);

const getSecretValue = {
  sk: 'testSK',
};

describe('functions', () => {
  describe('db', () => {
    describe('loadStripe', () => {
      afterEach(() => {
        getSecret.mockReset();
        global.console.log.mockReset();
      });

      it('fails and returns unexpected error, when getSecret throws error', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        getSecret.mockReturnValue(Promise.reject(error));

        try {
          await loadStripe();
        } catch (errorCatch) {
          expect(errorCatch).toEqual(error);
        }
      });

      it('fails and returns unexpected error, when getSecret does not return a value', async () => {
        getSecret.mockReturnValue(Promise.resolve());

        try {
          await loadStripe();
        } catch (errorCatch) {
          expect(errorCatch).toMatchSnapshot();
        }
      });

      it('correctly loads stripe', async () => {
        getSecret.mockReturnValue(Promise.resolve({ value: getSecretValue }));

        const result = await loadStripe();

        expect(result).toEqual(null);
        expect(global.stripe).toEqual(stripeObject);
        expect(global.console.log).toMatchSnapshot();
      });
    });
  });
});
