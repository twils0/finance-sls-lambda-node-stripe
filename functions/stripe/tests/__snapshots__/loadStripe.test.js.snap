// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions db loadStripe correctly loads stripe 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Stripe Loaded
",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`functions db loadStripe fails and returns unexpected error, when getSecret does not return a value 1`] = `[TypeError: Cannot destructure property \`value\` of 'undefined' or 'null'.]`;
